Define LinearRegressionRequest and Response messages

inf0rmatiker opened the pull request:
Title says it all

hamersaw commented on src/main/proto/sustain.proto:
```diff
@@ -64,11 +64,39 @@ message KMeansClusteringResponse {
 
 message LinearRegressionResponse {
-  // TODO - fill out
+  repeated Collection collections = 1;
```
I don't think we need this. If it's included on the Request than the client-side already has the information.

hamersaw commented on src/main/proto/sustain.proto:
```diff
@@ -64,11 +64,39 @@ message KMeansClusteringResponse {
 message LinearRegressionRequest {
-  // TODO - fill out
+  repeated Collection collections = 1;
+  repeated string gisJoins = 2;
```
Should we define this on the ModelRequest? It seems to be a field that could be used in all model training.

inf0rmatiker replied:
Agreed

hamersaw commented on src/main/proto/sustain.proto:
```diff
@@ -64,11 +64,39 @@ message KMeansClusteringResponse {
-  // TODO - fill out
+  repeated Collection collections = 1;
+
+  message LinearRegressionSummary {
+    string gisJoin = 1;
+    double totalIterations = 2;
+    double rmseResidual = 3;
+    double r2Residual = 4;
+    double intercept = 5;
+    repeated double slopeCoefficients = 6;
+    repeated double objectiveHistory = 7;
+  }
+  repeated LinearRegressionSummary summaries = 2;
```
The rpc definition for modelRequest is a stream. The LinearRegressionResponse could simply be the fields in the LinearRegressionSummary message as we will be returning multiple instances through rpc.

inf0rmatiker commented:
- Updated the proto definition of `LinearRegressionResponse` -- puts all fields at root level of object.
- Removed Collection from `LinearRegressionResponse` and moved it into `ModelRequest`, since it will be used by all model requests.

menuka94 commented on src/main/proto/sustain.proto:
```diff
@@ -64,11 +65,33 @@ message KMeansClusteringResponse {
+
+message Collection {
+  string name = 1;
+  string label = 2;
```
What does `label` represent?

inf0rmatiker replied:
This is the column, or feature we are trying to predict.
